src/SignalR/server/Core/src/HubConnectionContext.cs

```diff
@@ -76,8 +75,8 @@ public HubConnectionContext(ConnectionContext connectionContext, HubConnectionCo
 
         HubCallerContext = new DefaultHubCallerContext(this);
 
-        _systemClock = contextOptions.SystemClock ?? new SystemClock();
-        _lastSendTick = _systemClock.CurrentTicks;
+        _time = contextOptions.Time ?? TimeProvider.System;
+        _lastSendTime = _time.GetUtcNow().UtcTicks;
```

Comment:
We are explicitly not using a Utc based clock in SignalR

Reply:
Yeah... TimeProvider can't do that. I'll revert this.

Reply:
It should, [GetTimestamp()](https://source.dot.net/#System.Private.CoreLib/src/libraries/Common/src/System/TimeProvider.cs,907ce93a0566b0d9,references)

Reply:
Aren't timestamps a lot more expensive due to their higher granularity?

Reply:
Utc is a lot more expensive, but this is done mostly for correctness, not perf

|        Method |      Mean |     Error |    StdDev | Allocated |
|-------------- |----------:|----------:|----------:|----------:|
|        UtcNow | 17.854 ns | 0.3675 ns | 0.4648 ns |         - |
| StopwatchTime |  9.997 ns | 0.2137 ns | 0.1999 ns |         - |

Reply:
@tarekgh is the perf difference above inevitable?

Reply:
Typically, using UTC involves additional operations, making it more costly compared to GetTimestamp which simply reports the system's number. I anticipate UTC's performance to be inferior on NETFX since we've already optimized it for net core.